### PR TITLE
Merge VCR into master

### PR DESCRIPTION
**Pull request by abhinav**
So there are three major parts:

- Cassette handles persistence. Request-response pairs are recorded to cassettes and the cassette is queried via the `can_replay` and `replay` methods.
- VCRProxyService is a TChannel Thrift service that consumes a Cassette. It accepts `Request` objects, representing TChannel requests, and checks if the Cassette can replay it. If yes, it returns the Cassette's response. Otherwise, it makes a real TChannel call (assuming the chosen record mode allows it) to the remote service, records the response into a `Response` object, and returns it to the caller.
- Patcher is responsible for monkey patching. It consumes a client to the running VCRProxyService. It intercepts `TChannel.request` calls, converts them into `Request` objects, and calls the VCRProxyService. It converts the returned `Response` objects into TChannel responses.

### CODE REVIEW

**Comment by abhinav** [comment]
@breerly @blampe @junchaowu 

Feel free to ignore this if you've seen the other VCR PRs. This is just final merge into master with an update to the changelog.

**Comment by breerly** [comment]
@abhinav @blampe  @junchaowu think now's a good time to start squashing commits into fewer logical pieces before merging to master?

**Comment by breerly** [comment]
I haven't reviewed enough of this - I'll get at it today

**Comment by breerly** [comment]
decorator doesn't work - please fix & write integration test

File: tests/testing/vcr/strategies.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+from __future__ import absolute_import
+
+from hypothesis.strategies import (
```
Comment (blampe) [inline tests/testing/vcr/strategies.py]: Powerful :)

**Comment by abhinav** [comment]
@breerly Fixed

**Comment by blampe** [comment]
:shipit:

**Comment by abhinav** [comment]
Squashed into one commit and merged into master.

**Pull request merged**